src/linkplay/utils.py

```diff
@@ -66,6 +74,50 @@ async def session_call_api_ok(
         raise LinkPlayRequestException(f"Didn't receive expected OK from {endpoint}")
 
 
+async def call_tcpuart(host: str, cmd: str) -> str | None:
+    """Get the latest data from TCP UART service."""
+    LENC = format(len(cmd), '02x')
+    HED1 = '18 96 18 20 '
+    HED2 = ' 00 00 00 c1 02 00 00 00 00 00 00 00 00 00 00 '
+    CMHX = ' '.join(hex(ord(c))[2:] for c in cmd)
+    data = None
+    _LOGGER.debug("Sending to %s TCP UART command: %s", host, cmd)
+    try:
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            s.settimeout(API_TIMEOUT)
+            s.connect((host, TCPPORT))
+            s.send(bytes.fromhex(HED1 + LENC + HED2 + CMHX))
+            data = str(repr(s.recv(1024))).encode().decode("unicode-escape")
+
+        pos = data.find("AXX")
+        if pos == -1:
+            pos = data.find("MCU")
+
+        data = data[pos:(len(data)-2)]
+        _LOGGER.debug(
+            "Received from %s TCP UART command result: %s", host, data)
+        try:
+            s.close()
+        except:
+            pass
+
+    except socket.error as ex:
+        _LOGGER.debug("Error sending TCP UART command: %s with %s", cmd, ex)
+        data = None
+
+    return data
+
+
+async def call_tcpuart_json(host: str, cmd: str) -> dict[str, str]:
+    raw_response = await call_tcpuart(host, cmd)
+    if not raw_response:
+        return dict()
+    strip_start = raw_response.find('{')
```

Comment:
Not super happy with this.  The raw response (once decoded) looks like this (including trailing `&` character) :

```
AXX+MEA+DAT{ "title": "504253203130362E37464D204D656C626F75726E65", "artist": "", "album": "", "vendor": "", "url": "756E6B6E6F776E" }&
```

Open to other ideas here

Reply:
I'm pretty sure the prefix itself is variable length, depending on the endpoint called.

My first cut just did `raw_response[11:-1]` but I'm not convinced it would be robust enough to handle other response payloads

Reply:
I don't think we need to optimize this just yet by determining the opening bracket, so I'd just slice from the first `{` to `-1`